examples/llm/icl_eval/evaluate_model.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 MosaicML Examples authors
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import time
+
+import torch
+from composer.loggers import InMemoryLogger
+from composer.trainer import Trainer
+from model_loading import load_composer_model
+from omegaconf import DictConfig
+from omegaconf import OmegaConf as om
+
+from examples.common.builders import build_evaluators
+from examples.llm.src.tokenizer import TOKENIZER_REGISTRY
+
+
+def validate_cfg(eval_cfg):
+    assert eval_cfg.get('dataset_uri',
+                        None) is not None, 'dataset_uri is required!'
+    assert 'type' in eval_cfg
+    assert 'num_fewshot' in eval_cfg
+    assert 'batch_size' in eval_cfg
+    assert 'metrics' in eval_cfg
+    assert 'formatting_options' in eval_cfg
+    assert 'prompt_string' in eval_cfg.get('formatting_options')
+    assert 'example_delimiter' in eval_cfg.get('formatting_options')
+    assert 'continuation_delimiter' in eval_cfg.get('formatting_options')
+    assert 'label' in eval_cfg
+
+
+def get_evaluators_from_config(cfg):
+    tokenizer = TOKENIZER_REGISTRY[cfg.tokenizer.type](**cfg.tokenizer.args)
+    evaluators = []
+    logger_keys = []
+    for eval_cfg in cfg.icl_tasks:
+        validate_cfg(eval_cfg)
+        dataset_uri = eval_cfg.get('dataset_uri')
+        icl_task_type = eval_cfg.get('type')
+        num_fewshots = eval_cfg.get('num_fewshot')
+        batch_size = eval_cfg.get('batch_size')
+        metrics = list(eval_cfg.get('metrics'))
+        prompt_string = eval_cfg.get('formatting_options').get('prompt_string')
+        example_delimiter = eval_cfg.get('formatting_options').get(
+            'example_delimiter')
+        continuation_delimiter = eval_cfg.get('formatting_options').get(
+            'continuation_delimiter')
+        label = eval_cfg.get('label')
+        max_seq_len = cfg.tokenizer.args.max_seq_len
+
+        res = build_evaluators(label, icl_task_type, dataset_uri, tokenizer,
+                               batch_size, max_seq_len, prompt_string,
+                               example_delimiter, continuation_delimiter,
+                               metrics, num_fewshots)
+        evaluators.extend(res[0])
+        logger_keys.extend(res[1])
+
+    return evaluators, logger_keys
+
+
+if __name__ == '__main__':
+    yaml_path, args_list = sys.argv[1], sys.argv[2:]
+    with open(yaml_path) as f:
+        yaml_cfg = om.load(f)
+    cli_cfg = om.from_cli(args_list)
+    cfg = DictConfig(om.merge(yaml_cfg, cli_cfg))
+
+    composer_model = load_composer_model(cfg)
```

Comment:
should this code path also support HF model? Not sure if we're intending to keep this usage of this file as a script around

Reply:
Yes, but even when loading pretrained HF weights, the model is a `ComposerModel`, it is a `HuggingFaceModel` that we are gonna pass to the Trainer. I changed the var/function names a bit to make that clear that its **not** a normal HF model coming out of there.

Reply:
Ah right 👍 just didn't look carefully enough at what load_composer_model was